dev-requirements.txt

```
@@ -0,0 +1,2 @@
+pip==25.1.1
```

From what I could find pinning `pip` is not recommended. This is not a package for the project but rather the package manager which can have different versions based on the system where you run it. Instead I think we should pin `ruff` 
```suggestion
ruff==0.11.13
```